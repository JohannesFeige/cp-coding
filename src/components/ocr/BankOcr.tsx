import React, { useCallback, useState, useEffect } from 'react';
import { processBankOcr } from '../shared/calculation.service';

const BankOcr: React.FC = () => {
  const [file, setFile] = useState<string>('');
  const [value, setValue] = useState<string>('');

  const fileChangeHandler = useCallback((changeEvent: React.ChangeEvent<HTMLInputElement>) => {
    const { files } = changeEvent.target;
    if (files && files.length) {
      const reader = new FileReader();
      reader.onloadend = function() {
        setFile(reader.result as string);
      };
      reader.readAsText(files[0]);
    }
  }, []);

  // update result on file change
  useEffect(() => {
    setValue(processBankOcr(file));
  }, [file]);

  return (
    <React.Fragment>
      <div>
        <input type="file" onChange={fileChangeHandler} />
      </div>
      <div>
        <span>Raw:</span>
        <pre>{file}</pre>
      </div>
      <div>
        <span>Result: </span>
        <span>{value}</span>
      </div>
    </React.Fragment>
  );
};

export default BankOcr;
