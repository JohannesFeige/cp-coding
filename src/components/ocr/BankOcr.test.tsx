import React from 'react';
import { shallow } from 'enzyme';
import BankOcr from './BankOcr';

describe('BankOcr', () => {
  it('should render', () => {
    const wrapper = shallow(<BankOcr />);
    expect(wrapper).toMatchSnapshot();
  });
});
