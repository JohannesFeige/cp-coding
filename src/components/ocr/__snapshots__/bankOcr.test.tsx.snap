// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BankOcr should render 1`] = `
<Fragment>
  <div>
    <input
      onChange={[Function]}
      type="file"
    />
  </div>
  <div>
    <span>
      Raw:
    </span>
    <pre />
  </div>
  <div>
    <span>
      Result: 
    </span>
    <span />
  </div>
</Fragment>
`;
